### PR TITLE
Integrate rpm and deb build scripts.

### PR DESCRIPTION
**Pull request by interactivefiction**
Simple script to create an RPM build for gargoyle.  Tested on Fedora 29/30/31.

### CODE REVIEW

File: JamrulesRPM [hunk 1]
```diff
@@ -0,0 +1,313 @@
+# You will probably need to edit this file to get everything to compile.
```
Comment (cspiegel) [inline JamrulesRPM]: It looks like the only difference in this `Jamrules` file is that `USESDL` is disabled.  It'd be cleaner to use the original `Jamrules`, disabling `USESDL` on the command line, i.e.:

```
jam -sUSESDL=no
```
during the build step.  That way this file won't have to be kept in sync with the original.

File: gargoyle-buildrpm.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
+#!/bin/sh
```
Comment (cspiegel) [inline gargoyle-buildrpm.sh]: This script uses some non-POSIX stuff (e.g. `pushd`), so this should be `/bin/bash`.

File: gargoyle-buildrpm.sh [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+if [ -d "./build/dist" ]
+then
+    rm -rfv ./build/dist
+fi
+if [ -d "./build/linux.release" ]
+then
+    rm -rfv ./build/linux.release
+fi
+#rm -rfv ./build/dist
```
Comment (cspiegel) [inline gargoyle-buildrpm.sh]: These commented lines can be removed.

**Comment by cspiegel** [comment]
I don't have any experience with RPM, but I built and installed RPM from source and gave this script a shot, and it wound up successfully building an RPM file for me.  I can't test it, but obviously you have.  There are a few comments I have, and I may wind up with a couple more things after looking this over again, but I'm happy merging it in when it's ready.

File: gargoyle-buildrpm.sh (hunk 2 above)
Comment (interactivefiction) [reply gargoyle-buildrpm.sh]: Will do.

File: JamrulesRPM (hunk 1 above)
Comment (interactivefiction) [reply JamrulesRPM]: Thanks for this, very helpful, will make the change!

Comment (interactivefiction) [reply JamrulesRPM]: I attempted a build using the commandline approach you mention but the build system does not seem to honor the commandline flag.  Still getting SDL errors in the build.  Any thoughts?

File: gargoyle-buildrpm.sh (hunk 3 above)
Comment (interactivefiction) [reply gargoyle-buildrpm.sh]: Checked in changes.

File: gargoyle-buildrpm.sh (hunk 2 above)
Comment (interactivefiction) [reply gargoyle-buildrpm.sh]: Checked in changes.

File: JamrulesRPM (hunk 1 above)
Comment (interactivefiction) [reply JamrulesRPM]: Checked in changes.

**Comment by interactivefiction** [comment]
Added a script to build DEB packages.  Script uses only dpkg-* tools, no higher level wrappers etc.  Tested resulting DEB packages on Ubuntu 16.04, 18.04, and 19.10.

**Comment by cspiegel** [comment]
I'm attaching an RPM I created if you'd like to test; but it installs `/opt/rpm` (because my RPM is installed there), just as a note.

It looks like the script still disables SDL; in light of the conclusions of #358, should the scripts be updated to re-enable SDL?

**Comment by interactivefiction** [comment]
I have a VM I can try and install that RPM to.  Out of curiosity, what environment are you building it in?  Yes, the script has SDL disabled only because I did the script work before I figured out the necessary RPM deps for SDL :->  I still need to build RPM packages with SDL enabled and put them through testing on fresh OS for F29,30,31 to make certain all of the new SDL runtime dependencies are properly accounted for during RPM install.  Currently, I do not know if the automatic dep generation properly accounts for them.  It should in theory, but testing it to be sure is probably best.  As the script is, I do know for sure that RPMs built without SDL will indeed install and auto-magically grab the required deps to run even if they are not currently installed on the system.

**Comment by cspiegel** [comment]
> Out of curiosity, what environment are you building it in?

I'm on a custom Linux distribution, so there's no real useful way to ensure you have a compatible setup.

If you update the scripts to include SDL, I think there's no reason not to get them merged in.

**Comment by interactivefiction** [comment]
Sounds good, when I get some spare time I will test with SDL on various systems and let you know.

**Comment by cspiegel** [comment]
@interactivefiction A lot has changed in Gargoyle since you wrote these, the major change being the build system moving to CMake from Jam.

I rebased these changes onto the latest master and made some fixes to get them working with the changes.  I tested on Ubuntu 20.04 and Fedora 34.

Because I had to rebase onto a much later version, instead of trying to incorporate the fixes back into your branch, I created a new branch off of yours, rebased, and made a new pull request based on it: #540 

If you have any input regarding these new scripts, please let me know: I've never worked with either Debian or RPM build scripts before, so it's entirely possible I'm doing things in a less-than-ideal way.

**Comment by cspiegel** [comment]
Closing after merging #540.  Thank you!

**Pull request closed without merging**